python/lsst/daf/butler/configs/dimensions.yaml

```diff
@@ -269,6 +273,18 @@ elements:
           Counter for the observation within a larger sequence. Context
           of the sequence number is observatory specific. Can be
           a global counter or counter within day_obs.
+      -
+        name: seq_start
+        type: int
+        doc: >
+          The sequence number of the first exposure of the visit that contains
+          this exposure.
```

Comment:
I'd like to find a way to reword this; I know what you mean, but "the visit that contains this exposure" is not a well-defined thing, because there are in general two such visits.  Maybe something like "the as-observed visit" or "the standard visit"?

Reply:
My point was that "this visit" isn't well defined in the sense that there are multiple visit systems, and an exposure can be in different visits for different visit systems; I think you mean "the visit as defined by the script queue or whatever writes this exposure's header", but it's confusing that we use "visit" to mean both:
- a concept defined by the script queue/header/exposure metadata
- a concept defined by middleware/DefineVisitsTAsk, which may or may not care what the header says.

Reply:
The only safe definition for the doc string is "A sequence number associated in some manner with this visit". Everything else involves some assumptions as to how define-visit, in our case, is setting that seq_num field. The seq_num definition shouldn't need to know whether it's the first exposure in the visit, the last one, or a random number not related to any exposure. At Rubin we require that the seq_num is the first exposure in the visit and both visit definitions agree with what that value is.

Reply:
Yeah, I think that the best description here is probably one that avoids saying "visit" at all; maybe something like "The first 'seq_num' value in the set of exposures this one is part of."  It'd be nicer to have a non-overloaded word for this particular kind of "set of exposures" (can't call it a visit, can't call it a sequence, can't call it a group!), but just not saying visit is probably fine.

(Note that this is in the definition for `exposure`, not `visit`; I'm not sure if that might have been part of where we were miscommunicating.)

Reply:
You are indeed correct. I thought we were talking about `visit` since I was only looking at the four lines. Sorry.

If this is `seq_start` then... The definition is effectively instrument-specific but something like:

> Earliest sequence number that might be related to this exposure.

? And similarly for seq_end.

Reply:
The migration script, that I just added for v2, sets both `seq_start` and `seq_end` to the value of `seq_num` from the same record. Should it try to do something different and use min/max from related exposures (not sure if there is an easy way to do that)?